docs/src/maintain/index.md

```diff
@@ -9,6 +9,10 @@ eleventyNavigation:
 
 This guide is intended for those who work as part of the ESLint project team.
 
+## [How ESLint is Maintained](overview)
+
+Explains how ESLint is maintained, including information about team, governance, and funding.
```

Comment:
Governance is still under Contribute right now, do we want to move it as part of this PR?